blivet/formats/fs.py

```diff
@@ -386,7 +386,12 @@ def _getFormatOptions(self, options=None, do_labeling=False):
         argv.append(self.device)
         return argv
 
-    def doFormat(self, options=None):
+    def _preCreate(self, **kwargs):
+        super(FS, self)._preCreate(**kwargs)
+        if not self.mkfsProg:
+            return
```

Comment:
I agree. The reason I didn't do that is the Device classes don't and I wanted to keep them the same. The main difference is that it's okay to call setup on an already-active device or format, but it's not okay to call create on an already-existing device or format. That's why setup/teardown have a way to exit early while create/destroy do not.